types.go

```diff
@@ -15,54 +15,62 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by a Method to specify the objective-function
-// information needed at an x location.
-type EvaluationType uint
+// Operation represents the set of operations requested by Method at each
+// iteration. It is a bitmap of *Iteration and *Evaluation constants.
+type Operation uint64
 
-// Evaluation types can be composed together using the binary or operator, for
-// example 'FuncEvaluation | GradEvaluation' to evaluate both the function
-// value and the gradient.
+// Supported Operations. Individual operations must NOT be combined together by
+// the binary OR operator except for the various *Evaluations.
 const (
-	NoEvaluation   EvaluationType = 0
-	FuncEvaluation EvaluationType = 1 << (iota - 1)
+	// NoOperation does not specify any operation.
```

Comment:
I am not sure about the word "iteration" in this phrase. No sane Method has a reason to return NoOperation from Iterate(), but I can imagine that maybe perhaps it could want to return NoOperation from Init() if it want to just pass the control to Iterate(). At any case, NoOperation will lead to Iterate() being called again without any particular action apart from checking the limits and sending the loc to Recorder.